includes/payment-tokens/class-wc-stripe-becs-debit-payment-token.php

```diff
@@ -111,12 +111,10 @@ public function set_last4( $last4 ) {
 	 * @inheritDoc
 	 */
 	public function is_equal_payment_method( $payment_method ): bool {
-		if (
-			WC_Stripe_Payment_Methods::BECS_DEBIT === $payment_method->type
-			&& ( $payment_method->{WC_Stripe_Payment_Methods::BECS_DEBIT}->fingerprint ?? null ) === $this->get_fingerprint() ) {
-			return true;
+		if ( WC_Stripe_Payment_Methods::BECS_DEBIT !== $payment_method->type ) {
+			return false;
 		}
 
-		return false;
+		return ( $payment_method->{WC_Stripe_Payment_Methods::BECS_DEBIT}->fingerprint ?? null ) === $this->get_fingerprint();
```

Comment:
Also, might it be clearer to use the property directly?
```suggestion
		// Becs Debit uses the au_becs_debit property.
		return ( $payment_method->au_becs_debit->fingerprint ?? null ) === $this->get_fingerprint();
```